xapian-applications/omega/handler_libmwaw.cc

```diff
@@ -0,0 +1,264 @@
+		}
+		case 's': {
+		    PARSE_FIELD(start, eol, "subject", keywords);
+		    break;
+		}
+		case 't': {
+		    PARSE_FIELD(start, eol, "title", title);
+		    break;
+		}
+	    }
+	} else if ((end - start) > 8 && memcmp(start, "dcterms:", 8) == 0) {
+	    start += 8;
+	    PARSE_FIELD(start, eol, "available", keywords);
+	}
+    }
+}
+
+static bool
+extract_word(string& dump,
+	     string& title,
+	     string& keywords,
+	     string& author,
+	     string& error,
+	     RVNGFileStream* input)
+{
+    RVNGString content_dump;
+    RVNGTextTextGenerator content(content_dump, false);
+    if (MWAWDocument::parse(input, &content) == MWAWDocument::MWAW_R_OK) {
+	dump = content_dump.cstr();
+    } else {
+	error = "Libmwaw Error: Failed to extract text";
+	return false;
+    }
+
+    RVNGString metadata_dump;
+    RVNGTextTextGenerator metadata(metadata_dump, true);
+    if (MWAWDocument::parse(input, &metadata) == MWAWDocument::MWAW_R_OK) {
+	const char* data = metadata_dump.cstr();
+	size_t len = metadata_dump.size();
+	parse_metadata(data, len, author, title, keywords);
+    } else {
+	error = "Libmwaw Error: Failed to extract metadata";
+	return false;
+    }
+    return true;
+}
+
+static bool
+extract_spreadsheet(string& dump,
+		    string& title,
+		    string& keywords,
+		    string& author,
+		    string& error,
+		    string& pages,
+		    RVNGFileStream* input)
+{
+    RVNGStringVector pages_metadata;
+    RVNGTextSpreadsheetGenerator metadata(pages_metadata, true);
+    MWAWDocument::Result result = MWAWDocument::parse(input, &metadata);
+    if (result != MWAWDocument::MWAW_R_OK) {
+	error = "Libmwaw Error: Failed to extract metadata";
+	return false;
+    } else {
+	string meta_temp;
+	for (unsigned i = 0; i < pages_metadata.size(); ++i) {
+	    meta_temp.append(pages_metadata[i].cstr());
+	}
+	const char* data = meta_temp.c_str();
+	size_t len = meta_temp.size();
+	parse_metadata(data, len, author, title, keywords);
+    }
+
+    RVNGStringVector pages_content;
+    RVNGTextSpreadsheetGenerator content(pages_content, false);
+    result = MWAWDocument::parse(input, &content);
+    if (result != MWAWDocument::MWAW_R_OK) {
+	error = "Libmwaw Error: Failed to extract text";
+	return false;
+    } else {
+	for (unsigned i = 0; i < pages_content.size(); ++i) {
+	    dump.append(pages_content[i].cstr());
+	}
+	pages = pages_content.size();
+    }
+    return true;
+}
+
+static bool
+extract_presentation(string& dump,
+		     string& pages,
+		     string& error,
+		     RVNGFileStream* input)
+{
+    RVNGStringVector pages_content;
+    RVNGTextPresentationGenerator content(pages_content);
+    MWAWDocument::Result result = MWAWDocument::parse(input, &content);
+    if (result != MWAWDocument::MWAW_R_OK) {
+	error = "Libmwaw Error: Failed to extract text";
+	return false;
+    } else {
+	for (unsigned i = 0; i < pages_content.size(); ++i) {
+	    dump.append(pages_content[i].cstr());
+	}
+	pages = pages_content.size();
+    }
```

Comment:
For the same reasons I think I pointed out in another PR, I'd get rid of the `else` here and put its contents at the top level of the function.  Same below.

Reply:
I have made these changes.